src/store/connection/ConnectionState.ts

```diff
@@ -68,6 +70,8 @@ export class ConnectionState<T extends TConnection = TConnection> {
     return [connection.sourceBlockId, connection.targetBlockId].join(":");
   }
 
+  public viewComponent: BaseConnection<TBaseConnectionProps, TBaseConnectionState, TGraphLayerContext, T>;
```

Comment:
Don't you want to make it private? 
And, as in BlockState, make a getViewComponent function?